ratelimit_test.go

```diff
@@ -13,7 +13,7 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-func ExampleRatelimit() {
+func Example() {
```

Comment:
what should we do if we want to add more examples? e.g. with different implementations etc.

Reply:
See commit message - we can do more thing by namespacing to struct/method, but also just appending suffixes, like `Example_One`, `Example_Two`.

Reply:
Yeah, you can rename this when you add more examples.